Release 0.0.4

stolarczyk opened the pull request:
This contribution makes changes to the `Project` object initialization to match the overall concept (the subproject can be specified) and adds backward compatibility to the `implied/derived_columns` (`implied/derived_attributes` now)

nsheff committed:
fix typos

stolarczyk committed:
Merge branch 'master' of https://github.com/pepkit/pepr into dev

stolarczyk committed:
Add derived implied and subannoation for subprojects

stolarczyk committed:
Remove obsolete function

stolarczyk committed:
Add tests and testing related changes to the code

stolarczyk committed:
Work on tests

stolarczyk committed:
Add subprojects vignette. Closes #8

stolarczyk committed:
Update README.md

no slot accessors

stolarczyk committed:
Fix #9

stolarczyk committed:
Merge branch 'dev' of https://github.com/pepkit/pepr into dev

stolarczyk committed:
Change derived columns to derived attributes, add backward compatilibity, remove default derived attribute

stolarczyk committed:
delete vignettes *Columns

stolarczyk committed:
Change to implied attributes, add backwards compatibility

stolarczyk committed:
Start working on PEP-SummarizedExperiment interface

stolarczyk committed:
Add docs and mwthods to interface class

stolarczyk committed:
Redisign the interface class completely, it is the extension of the SummarizedExperiment now

stolarczyk committed:
Add vignette, reorganize dir, change .Rbuildignore

stolarczyk committed:
New approach to interface class implementation, needs work

stolarczyk committed:
Further work on multiple inheritance in interface class

stolarczyk committed:
Add methods to multiple inheritance interface class

stolarczyk committed:
Update vignette

stolarczyk committed:
Add constants support

stolarczyk committed:
update .Rbuildignore

stolarczyk committed:
Add vignette for constants and related changes/fixes

stolarczyk committed:
Unify interface of the methods and functions in initialize method

stolarczyk committed:
Remove classes definition; in separate repo now

stolarczyk committed:
Change printing method to cat

stolarczyk committed:
Allow for empty Project creation

stolarczyk committed:
fixed issues with derived and implied attributes backwards compatibility and displaying

stolarczyk committed:
change activateSubproject() to method + docs

stolarczyk committed:
add the subproject slot to the Project object. The subprojects can be activated during the object constuction now

nsheff committed:
version bump on dev

nsheff commented:
wait -- why do we need a `subproject` slot on the `Project` object? I don't follow...

stolarczyk committed:
remove the subproject slot from the Project class

stolarczyk commented:
You're right, we don't need it. It was the easiest way to pass the subproject name to the `initialize()` method and that's why I did it. Fixed it in: https://github.com/pepkit/pepr/commit/5612340db613c7537b9df2de047afae5e4d5b358

stolarczyk committed:
Add changelog. Closes #12

stolarczyk committed:
Work on docs examples

stolarczyk committed:
Change samples method. Closes #13

nsheff commented on R/loadConfig.R:
```diff
@@ -69,55 +71,17 @@ loadConfig = function(filename=NULL, sp=NULL) {
 }
 
 
-#' Mapper of organism name to genomic assembly name
```
What were these assembly by organism functions written for? was this written by @vreuter ?

vreuter replied:
I wrote something similar to this for `peppy`/`looper` but don't remember doing so for this package

stolarczyk replied:
It wasn't me either

stolarczyk committed:
Fix #14

The expandPath function redesign, the non-path strings remain untouched now

stolarczyk committed:
update docs

nsheff reviewed (changes requested):
Please keep line length within bioc specificiations

nsheff commented on R/loadConfig.R:
```diff
@@ -69,88 +71,69 @@ loadConfig = function(filename=NULL, sp=NULL) {
-#' @seealso \url{http://looper.readthedocs.io/en/latest/implied-columns.html}
+#' Expand system path
+#' 
+#' This function expands system paths (the non-absolute paths become absolute) and replaces the enviornment variables (e.g, \code{${HOME}}) with their vaules.
```
spelling check here, environment; values -- and line length. if you use an IDE that  that can wrap lines that can help

nsheff commented:
Looking good @michalstolarczyk  -- are you ready to merge this (after my minor style change suggestions)?

nsheff commented:
Can you make sure to pass `R CMD check` on this?

stolarczyk committed:
code cleanup

change length of code lines, docs, changelog

stolarczyk commented:
Ok, made the changes to the code, package passes the `R CMD check`. Let's merge

nsheff committed:
Merge github.com:pepkit/pepr into dev

nsheff committed:
Merge branch 'dev' of github.com:pepkit/pepr into dev

nsheff committed:
remove extra notes file

stolarczyk committed:
update package data

stolarczyk committed:
Fix expandPath function. Closes #14

nsheff committed:
expand intro vignette. Fix #16

nsheff committed:
Merge branch 'dev' of github.com:pepkit/pepr into dev

stolarczyk committed:
update pkgdown site vignettes section. Closes #15

stolarczyk committed:
add the pkgdown site to the package repository

nsheff commented:
Don't put the pkgdown site in the package repository; we host them at code.databio.org

stolarczyk committed:
rename vignettes for slick organization. References #15

stolarczyk committed:
update pkgdown config to work with new vignette names. #15

stolarczyk committed:
delete docs

stolarczyk committed:
fix typos in roxygen comments

nsheff committed:
Update README.md

nsheff committed:
change vignette group titles

nsheff committed:
Update gettingStarted.Rmd

nsheff committed:
Update gettingStarted.Rmd

nsheff committed:
update strformat docs

maybe fixes #19 ?

nsheff committed:
roxygenize

nsheff committed:
update docs

nsheff committed:
update pkgdown titles

nsheff committed:
update readme

nsheff committed:
update to BSD-2

nsheff committed:
add pep shield

stolarczyk committed:
make expandPath replace bracketed, not bracketed and mix of env vars. Related to #19

stolarczyk committed:
Merge branch 'dev' of https://github.com/pepkit/pepr into dev

stolarczyk committed:
remove warning, fix #18

stolarczyk committed:
add support for Project object in listSubproject fun

resolves #20

stolarczyk committed:
add support for multiple values in the YAML key-value pairs. See #21

stolarczyk committed:
update vignettes, see #17

stolarczyk committed:
Fix vignettes broke links, work on docs. Closes #17

stolarczyk committed:
update feature name to constantAttributes

stolarczyk committed:
add tests for multiple yaml values for one key, work on docs, update examples

stolarczyk committed:
organize tests in contexts

stolarczyk committed:
backwards compatibility in constantAttributes

stolarczyk committed:
muliple changes

change function names (fixing exporting), add getSample function, reorganize vignettes

stolarczyk committed:
fix docs

stolarczyk committed:
update pkgdown config

nsheff committed:
Minor language changes to web

stolarczyk committed:
add changelog to the pkgdown site, small changes to Project display

stolarczyk committed:
addresses #22

change NULL values to NA in while implying attributes,
pkgdown website layout changes
